content/recipes/interoperability/registering-swisslipids-identifiers.md

```diff
@@ -0,0 +1,162 @@
+
+The recipe uses [Bacting](https://github.com/egonw/bacting) {footcite}`Willighagen2021` which can be used in Groovy and Python. We
+will use the first here. The starting point is mappings (tuples) that link the SwissLipid identifiers to InChIKeys. The latter
+is used to find the matching Wikidata items.
+
+### Step 1: getting the data
+
+The data is CC-BY but it was agreed that adding the SwissLipid identifiers to Wikidata (CCZero) is okay.
+
+* Download `lipids.tsv` from the [Downloads](https://www.swisslipids.org/#/downloads) page
+* Gunzip the file
+
+### Step 2: extract Swiss Lipid ID <> InChIKey tuples
+
+For this step I use `csvtool` (`apt get install csvtool`):
+
+```shell
+csvtool -t TAB col 1,11 swisslipids.tsv
+```
+
+The output needs some further clean up, like removing lines without InChIKeys or "none" and "-" as value. Also,
+the "InChIKey=" prefix is removed in preparation for the next step. The full used code is:
+
+```shell
+csvtool -t TAB col 1,11 swisslipids.tsv  | sed 's/InChIKey=//' | grep -v "none" | grep -v ",-$" | grep -v ",$" | tee swisslipids_ids.tsv
+```
+
+This results in almost 600k tuples:
+
+```shell
+$ wc -l swiss*tsv
+   592412 swisslipids_ids.tsv
+   777957 swisslipids.tsv
+```
+
+### Step 3: creating a ShEx model
+
+Skipping this step for later this week, but here the task is to create a shape expression for Wikidata, to model how
+the identifiers will be added to Wikidata. See the recept publication by Waagmeester et al. about _A protocol for adding knowledge to Wikidata: aligning resources on human coronaviruses_
+{footcite}`Waagmeester2021`.
+
+### Step 4: creating QuickStatements
+
+Now we have the mappings and the data model in Wikidata, we can create QuickStatements to allow us to enter the
+data into Wikidata. This is not the only approach, and the process can be further automated using "Wikidata bots".
+For this, see BioHackathon Europe [Project 32](https://github.com/elixir-europe/biohackathon-projects-2021/tree/main/projects/32):
+Connecting ELIXIR-related open data on Wikidata via [WikiProject ELIXIR](https://www.wikidata.org/wiki/Wikidata:WikiProject_Elixir).
+
+Based on existing Bacting scripts, a script is created to take the `swisslipids_ids.tsv` file as input and create
+QuickStatements: [https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy](https://github.com/egonw/ons-wikidata/blob/master/ExtIdentifiers/swisslipids.groovy) This script is using [Apache Groovy](http://www.groovy-lang.org/)
+but Bacting can also be using in Python, see [https://github.com/cthoyt/pybacting](https://github.com/cthoyt/pybacting).
+
+This script uses a federated query against https://beta.sparql.swisslipids.org/sparql/ after a suggestion by Dr [Jerven Bolleman](https://orcid.org/0000-0002-7449-1266)
+who indicated that the [RDF4J](https://rdf4j.org/) backing this SPARQL endpoint will automatically batch the query against Wikidata, overcoming
+limitations of the Wikidata Query Service:
```

Comment:
````suggestion
```{note}
This script uses a federated query against https://beta.sparql.swisslipids.org/sparql/ after a suggestion by Dr [Jerven Bolleman](https://orcid.org/0000-0002-7449-1266)
who indicated that the [RDF4J](https://rdf4j.org/) backing this SPARQL endpoint will automatically batch the query against Wikidata, overcoming
limitations of the Wikidata Query Service:
```
````